cmd/juju/application/refresh.go

```diff
@@ -353,17 +365,35 @@ func (c *refreshCommand) Run(ctx *cmd.Context) error {
 	defer func() { _ = apiRoot.Close() }()
 
 	charmRefreshClient := c.NewCharmRefreshClient(apiRoot)
-	oldURL, oldOrigin, err := charmRefreshClient.GetCharmURLOrigin(ctx, c.ApplicationName)
-	if err != nil {
-		return errors.Trace(err)
-	}
 
 	// There is a timing window where deploy has been called and the charm
 	// is not yet downloaded. Check here to verify the origin has an ID,
-	// otherwise refresh result may be in accurate. We could use the
-	// retry package, but the issue is only seen in automated test due to
-	// speed. Can always use retry if it becomes an issue.
-	if oldOrigin.Source == commoncharm.OriginCharmHub && oldOrigin.ID == "" {
+	// otherwise refresh result may be in accurate.
+	// It should only occur in automated test, but if the default retries isn't enough,
+	// the returned error should help to investigate.
+	// Outside automated test, a new refresh will end up to works
+	var (
+		oldURL    *charm.URL
+		oldOrigin commoncharm.Origin
+	)
+
+	err = retry.Call(retry.CallArgs{
```

Comment:
Maybe I'm wrong (I haven't dive deep into the tests) but we could test this new behavior right?

Reply:
There is not real changes in behavior, from the user perspective (it only induce a small delay in case of retry, but the change is totally silent for user).

Since I don't add flags to change the retry count, neither the retry delay, I don't think it worth testing. 

Moreover, retry is tested as a whole in the dependency.

I don't think it worth a TU (and to actually test that their is retry it would imply several changes, either in retry loop - to catch the number of retry and return a trackable error, and in the TU - to have a mock that may fail).

What do you think ?

Reply:
Yes OK, agreed.